From 8969e907bc7c9d4be684f0ddcdcf714ef23587c5 Mon Sep 17 00:00:00 2001
From: Yassin <yassin@Nixos>
Date: Mon, 24 Aug 2026 16:03:08 +0200
Subject: [PATCH 1/9] dt-bindings: mediatek: describe MT6765 boot hardware

Describe the MT6765 peripherals needed to boot from eMMC and provide
USB device mode, including pinctrl and infracfg reset support. Add the
Lenovo TB-X606F board compatible and allow the existing MT6357 ADC
child.
---
 .../devicetree/bindings/arm/mediatek.yaml     |  1 +
 .../bindings/clock/mediatek,infracfg.yaml     |  1 +
 .../bindings/mfd/mediatek,mt6357.yaml         |  6 ++
 .../devicetree/bindings/mmc/mtk-sd.yaml       |  1 +
 .../bindings/phy/mediatek,tphy.yaml           |  1 +
 .../pinctrl/mediatek,mt6765-pinctrl.yaml      | 91 +++++++++++++++++++
 .../bindings/usb/mediatek,musb.yaml           |  1 +
 include/dt-bindings/clock/mt6765-clk.h        |  3 +-
 .../reset/mediatek,mt6765-infracfg.h          |  8 ++
 9 files changed, 112 insertions(+), 1 deletion(-)
 create mode 100644 Documentation/devicetree/bindings/pinctrl/mediatek,mt6765-pinctrl.yaml
 create mode 100644 include/dt-bindings/reset/mediatek,mt6765-infracfg.h

diff --git a/Documentation/devicetree/bindings/arm/mediatek.yaml b/Documentation/devicetree/bindings/arm/mediatek.yaml
index 382d0eb4d..2a02ee88a 100644
--- a/Documentation/devicetree/bindings/arm/mediatek.yaml
+++ b/Documentation/devicetree/bindings/arm/mediatek.yaml
@@ -57,6 +57,7 @@ properties:
       - items:
           - enum:
               - mediatek,mt6765-evb
+              - lenovo,tb-x606f
           - const: mediatek,mt6765
       - items:
           - enum:
diff --git a/Documentation/devicetree/bindings/clock/mediatek,infracfg.yaml b/Documentation/devicetree/bindings/clock/mediatek,infracfg.yaml
index d1d30700d..afeb2230c 100644
--- a/Documentation/devicetree/bindings/clock/mediatek,infracfg.yaml
+++ b/Documentation/devicetree/bindings/clock/mediatek,infracfg.yaml
@@ -65,6 +65,7 @@ if:
         enum:
           - mediatek,mt2701-infracfg
           - mediatek,mt2712-infracfg
+          - mediatek,mt6765-infracfg
           - mediatek,mt6795-infracfg
           - mediatek,mt7622-infracfg
           - mediatek,mt7986-infracfg
diff --git a/Documentation/devicetree/bindings/mfd/mediatek,mt6357.yaml b/Documentation/devicetree/bindings/mfd/mediatek,mt6357.yaml
index b67fbe0e7..31564b65b 100644
--- a/Documentation/devicetree/bindings/mfd/mediatek,mt6357.yaml
+++ b/Documentation/devicetree/bindings/mfd/mediatek,mt6357.yaml
@@ -16,6 +16,7 @@ description: |
   USB battery charging, fuel gauge, RTC
 
   This is a multifunction device with the following sub modules:
+  - Auxiliary ADC
   - Regulator
   - RTC
   - Keys
@@ -55,6 +56,11 @@ properties:
            2100000, 2500000, 2600000, 2700000]
     default: 1700000
 
+  adc:
+    $ref: /schemas/iio/adc/mediatek,mt6359-auxadc.yaml
+    unevaluatedProperties: false
+    description: PMIC auxiliary analog-to-digital converter.
+
   regulators:
     type: object
     $ref: /schemas/regulator/mediatek,mt6357-regulator.yaml
diff --git a/Documentation/devicetree/bindings/mmc/mtk-sd.yaml b/Documentation/devicetree/bindings/mmc/mtk-sd.yaml
index eb3755bdf..20dcd0e10 100644
--- a/Documentation/devicetree/bindings/mmc/mtk-sd.yaml
+++ b/Documentation/devicetree/bindings/mmc/mtk-sd.yaml
@@ -16,6 +16,7 @@ properties:
       - enum:
           - mediatek,mt2701-mmc
           - mediatek,mt2712-mmc
+          - mediatek,mt6765-mmc
           - mediatek,mt6779-mmc
           - mediatek,mt6795-mmc
           - mediatek,mt7620-mmc
diff --git a/Documentation/devicetree/bindings/phy/mediatek,tphy.yaml b/Documentation/devicetree/bindings/phy/mediatek,tphy.yaml
index ff5c77ef1..99a2504ae 100644
--- a/Documentation/devicetree/bindings/phy/mediatek,tphy.yaml
+++ b/Documentation/devicetree/bindings/phy/mediatek,tphy.yaml
@@ -70,6 +70,7 @@ properties:
     oneOf:
       - items:
           - enum:
+              - mediatek,mt6765-tphy
               - mediatek,mt2701-tphy
               - mediatek,mt7623-tphy
               - mediatek,mt7622-tphy
diff --git a/Documentation/devicetree/bindings/pinctrl/mediatek,mt6765-pinctrl.yaml b/Documentation/devicetree/bindings/pinctrl/mediatek,mt6765-pinctrl.yaml
new file mode 100644
index 000000000..243471acc
--- /dev/null
+++ b/Documentation/devicetree/bindings/pinctrl/mediatek,mt6765-pinctrl.yaml
@@ -0,0 +1,91 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/pinctrl/mediatek,mt6765-pinctrl.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: MediaTek MT6765 Pin Controller
+
+maintainers:
+  - Sean Wang <sean.wang@kernel.org>
+
+properties:
+  compatible:
+    const: mediatek,mt6765-pinctrl
+
+  reg:
+    items:
+      - description: GPIO controller registers
+      - description: IO configuration bank 1
+      - description: IO configuration bank 2
+      - description: IO configuration bank 3
+      - description: IO configuration bank 4
+      - description: IO configuration bank 5
+      - description: IO configuration bank 6
+      - description: IO configuration bank 7
+      - description: External interrupt controller registers
+
+  reg-names:
+    items:
+      - const: iocfg0
+      - const: iocfg1
+      - const: iocfg2
+      - const: iocfg3
+      - const: iocfg4
+      - const: iocfg5
+      - const: iocfg6
+      - const: iocfg7
+      - const: eint
+
+  gpio-controller: true
+
+  '#gpio-cells':
+    const: 2
+
+  gpio-ranges:
+    maxItems: 1
+
+  interrupt-controller: true
+
+  '#interrupt-cells':
+    const: 2
+
+  interrupts:
+    maxItems: 1
+
+patternProperties:
+  '-pins$':
+    type: object
+    additionalProperties: false
+    patternProperties:
+      '^pins':
+        type: object
+        $ref: /schemas/pinctrl/pincfg-node.yaml
+        properties:
+          pinmux: true
+
+          bias-pull-up: true
+
+          bias-pull-down: true
+
+          input-enable: true
+
+          drive-strength:
+            minimum: 0
+            maximum: 7
+        required:
+          - pinmux
+        additionalProperties: false
+
+required:
+  - compatible
+  - reg
+  - reg-names
+  - gpio-controller
+  - '#gpio-cells'
+  - gpio-ranges
+  - interrupt-controller
+  - '#interrupt-cells'
+  - interrupts
+
+additionalProperties: false
diff --git a/Documentation/devicetree/bindings/usb/mediatek,musb.yaml b/Documentation/devicetree/bindings/usb/mediatek,musb.yaml
index a39d38db7..3b4425ff2 100644
--- a/Documentation/devicetree/bindings/usb/mediatek,musb.yaml
+++ b/Documentation/devicetree/bindings/usb/mediatek,musb.yaml
@@ -17,6 +17,7 @@ properties:
   compatible:
     items:
       - enum:
+          - mediatek,mt6765-musb
           - mediatek,mt8516-musb
           - mediatek,mt2701-musb
           - mediatek,mt7623-musb
diff --git a/include/dt-bindings/clock/mt6765-clk.h b/include/dt-bindings/clock/mt6765-clk.h
index eb97e5685..4df36b57e 100644
--- a/include/dt-bindings/clock/mt6765-clk.h
+++ b/include/dt-bindings/clock/mt6765-clk.h
@@ -225,7 +225,8 @@
 #define CLK_IFR_CCIF2_MD		61
 #define CLK_IFR_CCIF3_AP		62
 #define CLK_IFR_CCIF3_MD		63
-#define CLK_IFR_NR_CLK			64
+#define CLK_IFR_PMIC_AP			64
+#define CLK_IFR_NR_CLK			65
 
 /* AUDIO */
 #define CLK_AUDIO_AFE			0
diff --git a/include/dt-bindings/reset/mediatek,mt6765-infracfg.h b/include/dt-bindings/reset/mediatek,mt6765-infracfg.h
new file mode 100644
index 000000000..afad8825d
--- /dev/null
+++ b/include/dt-bindings/reset/mediatek,mt6765-infracfg.h
@@ -0,0 +1,8 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+
+#ifndef _DT_BINDINGS_RESET_MT6765_INFRACFG_H
+#define _DT_BINDINGS_RESET_MT6765_INFRACFG_H
+
+#define MT6765_INFRA_RST2_PMIC_WRAP	0
+
+#endif
